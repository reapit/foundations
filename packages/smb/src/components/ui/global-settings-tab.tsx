import * as React from 'react'
export type GlobalSettingsTabProps = {}

export const GlobalSettingsTab: React.FC<GlobalSettingsTabProps> = () => null

export default GlobalSettingsTab
