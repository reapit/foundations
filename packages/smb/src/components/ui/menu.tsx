import * as React from 'react'
import { withRouter, RouteComponentProps } from 'react-router'
import { Menu as Sidebar, MenuConfig, ReapitLogo } from '@reapit/elements'
import { LoginMode } from '@reapit/cognito-auth'
import { Location } from 'history'
import { FaSignOutAlt, FaCloud } from 'react-icons/fa'

export const generateMenuConfig = (
  logoutCallback: () => void,
  location: Location<any>,
  mode: LoginMode,
): MenuConfig => {
  return {
    defaultActiveKey: 'LOGO',
    mode,
    location,
    menu: [
      {
        key: 'LOGO',
        icon: <ReapitLogo className="nav-item-icon" />,
        type: 'LOGO',
      },
      {
        title: 'Apps',
        key: 'APPS',
        icon: <FaCloud className="nav-item-icon" />,
        callback: () =>
          (window.location.href =
            window.location.href.includes('dev') || window.location.href.includes('localhost')
              ? 'https://dev.marketplace.reapit.cloud/client/installed'
              : 'https://marketplace.reapit.cloud/client/installed'),
        type: 'PRIMARY',
      },
      {
        title: 'Logout',
        key: 'LOGOUT',
        callback: logoutCallback,
        icon: <FaSignOutAlt className="nav-item-icon" />,
        type: 'SECONDARY',
      },
    ],
  }
}

export type DispatchProps = {
  logout: () => void
}

export type StateProps = {
  mode: LoginMode
}

export type MenuProps = DispatchProps & StateProps & RouteComponentProps & {}

export const logoutCallback = logout => () => logout()

export const Menu: React.FC<MenuProps> = ({ logout, location, mode }: MenuProps) => {
  const menuConfigs = generateMenuConfig(logoutCallback(logout), location, mode)
  return <Sidebar {...menuConfigs} location={location} />
}

export default withRouter(Menu) as any
