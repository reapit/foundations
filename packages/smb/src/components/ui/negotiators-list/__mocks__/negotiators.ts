import { NegotiatorsQueryResponse } from '../negotiators-list'

export const negotiators: NegotiatorsQueryResponse = {
  GetNegotiators: {
    _embedded: [
      {
        id: 'MGL',
        created: '2014-12-26T12:22:01.0000000Z',
        modified: '2016-11-29T09:57:32.0000000Z',
        name: 'Abel Robertson',
        active: true,
        officeId: 'NPG',
        email: 'abel.robertson@reapitestates.net',
        metadata: {},
        _eTag: '"10109C0209C684789B72FFC53730E31C"',
        _links: {
          self: {
            href: '/negotiators/MGL',
          },
          office: {
            href: '/offices/NPG',
          },
        },
      },
      {
        id: 'RPA',
        created: '2017-02-07T12:08:53.0000000Z',
        modified: '2017-02-07T12:09:41.0000000Z',
        name: 'Accounts User',
        active: true,
        officeId: 'MCL',
        _eTag: '"85A9DB571DB893A5FB734105AFF6B464"',
        _links: {
          self: {
            href: '/negotiators/RPA',
          },
          office: {
            href: '/offices/MCL',
          },
        },
      },
      {
        id: 'KLB',
        created: '2010-11-15T14:53:08.0000000Z',
        modified: '2017-03-10T00:02:08.0000000Z',
        name: 'Adele Small',
        active: true,
        officeId: 'NGL',
        email: 'adele.small@reapitestates.net',
        metadata: {},
        _eTag: '"B28EE5CFF654DB22BF3247CA028C10C3"',
        _links: {
          self: {
            href: '/negotiators/KLB',
          },
          office: {
            href: '/offices/NGL',
          },
        },
      },
    ],
    pageNumber: 1,
    pageSize: 3,
    pageCount: 3,
    totalCount: 3,
    _links: {
      self: {
        href: '/negotiators/?PageNumber=1&PageSize=3',
      },
      first: {
        href: '/negotiators/?PageNumber=1&PageSize=3',
      },
      next: {
        href: '/negotiators/?PageNumber=2&PageSize=3',
      },
      last: {
        href: '/negotiators/?PageNumber=3&PageSize=3',
      },
    },
  },
}
