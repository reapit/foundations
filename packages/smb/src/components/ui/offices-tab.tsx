import * as React from 'react'
import { Spreadsheet } from '@reapit/elements'

const dataBasic = [
  [
    { readOnly: true, value: 'Office Name' },
    { readOnly: true, value: 'Building Name' },
    { readOnly: true, value: 'Building No.' },
    { readOnly: true, value: 'Address 1' },
    { readOnly: true, value: 'Address 2' },
    { readOnly: true, value: 'Address 3' },
    { readOnly: true, value: 'Address 4' },
    { readOnly: true, value: 'Post Code' },
    { readOnly: true, value: 'Telephone' },
    { readOnly: true, value: 'Fax' },
    { readOnly: true, value: 'Email' },
  ],
  [
    { value: 'London' },
    { value: 'The White House' },
    { value: '15' },
    { value: 'London 1' },
    { value: '' },
    { value: 'Londom 3' },
    { value: '' },
    { value: 'EC12NH' },
    { value: '0845 0000' },
    { value: '' },
    { value: 'row1@gmail.com' },
  ],
  [
    { value: 'London2' },
    { value: 'The Black House' },
    { value: '11' },
    { value: '' },
    { value: '' },
    { value: 'Adress 3' },
    { value: '' },
    { value: 'EC12NH' },
    { value: '087 471 929' },
    { value: '' },
    { value: 'row2@gmail.com' },
  ],
  [
    { value: 'New York' },
    { value: 'Building A' },
    { value: '11' },
    { value: '' },
    { value: '' },
    { value: 'City Z' },
    { value: '' },
    { value: 'AL7187' },
    { value: '017 7162 9121' },
    { value: '' },
    { value: 'row3@gmail.com' },
  ],
]

export type OfficesTabProps = {}

export const OfficesTab: React.FC<OfficesTabProps> = () => {
  return (
    <div>
      <Spreadsheet
        data={dataBasic}
        description={
          <p>
            <p>
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the
              industry&apos;s standard dummy text ever since the 1500s, when an unknown printer took a galley of type
              and scrambled it to make a type specimen book.
            </p>
          </p>
        }
      />
    </div>
  )
}

export default OfficesTab
