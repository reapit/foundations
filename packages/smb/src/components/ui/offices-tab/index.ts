import OfficesTab from './offices-tab'

export default OfficesTab
