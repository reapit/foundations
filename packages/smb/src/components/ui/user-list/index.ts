import UserList from './user-list'
export default UserList
