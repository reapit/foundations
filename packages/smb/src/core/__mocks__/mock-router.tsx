import { RouteComponentProps } from 'react-router'
import { UnregisterCallback, Href } from 'history'

// This is to mock out the dependencies for react router
export function getMockRouterProps<P>({ params, search = '' }: { params: P; search: string }) {
  const location = {
    hash: 'mockHash',
    key: 'mockKey',
    pathname: 'mockPathname',
    search: search,
    state: {},
  }

  const props: RouteComponentProps<P | {}> = {
    match: {
      isExact: true,
      params: params || {},
      path: '/mockPathname',
      url: '/mockPathname',
    },
    location: location,
    history: {
      length: 2,
      action: 'POP',
      location: location,
      push: jest.fn(),
      replace: jest.fn(),
      go: jest.fn(),
      goBack: jest.fn(),
      goForward: jest.fn(),
      block: () => {
        // tslint:disable-next-line:no-empty
        const temp: UnregisterCallback = () => {}
        return temp
      },
      createHref: () => {
        const temp: Href = ''
        return temp
      },
      listen: () => {
        // tslint:disable-next-line:no-empty
        const temp: UnregisterCallback = () => {}
        return temp
      },
    },
    staticContext: {},
  }

  return props
}
