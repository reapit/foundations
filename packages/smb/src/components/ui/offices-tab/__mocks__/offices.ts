import { OfficeModel } from '@reapit/foundations-ts-definitions'
import { OfficesQueryResponse, CreateOfficeParams, UpdateOfficeParams } from '../offices-tab'
import { ChangedCells } from '@reapit/elements/src/components/Spreadsheet/types'

export const offices: OfficesQueryResponse = {
  GetOffices: {
    _embedded: [
      {
        id: 'REA',
        created: '2020-02-28T10:34:06.0000000Z',
        modified: '2020-02-28T10:34:06.0000000Z',
        name: 'Reapit',
        manager: 'Mr John Smith',
        address: {
          buildingName: '',
          buildingNumber: '15',
          line1: 'Example street',
          line2: 'Solihull',
          line3: 'West Midlands',
          line4: '',
          postcode: 'B91 2XX',
          countryId: 'GB',
        },
        workPhone: '01234 567890',
        email: 'example@email.com',
        metadata: {},
        _eTag: '"104F6D31FAFEB3B1DE6BB9CF8E071094"',
        _links: { self: { href: '/offices/REA' }, negotiators: { href: '/negotiators/?officeId=REA' } },
      },
      {
        id: 'STL',
        created: '2016-12-01T15:32:45.0000000Z',
        modified: '2020-02-14T15:52:06.0000000Z',
        name: 'Stevenage Lettings',
        manager: '',
        address: {
          buildingName: '',
          buildingNumber: '',
          line1: "St George's Way",
          line2: 'Stevenage',
          line3: 'Hertfordshire',
          line4: '',
          postcode: 'SG1 1XB',
          countryId: 'GB',
        },
        workPhone: undefined,
        email: 'lettings@hughmason.co.uk',
        metadata: {},
        _eTag: '"17465C0780A21582FFE1B06E35DEE37D"',
        _links: { self: { href: '/offices/STL' }, negotiators: { href: '/negotiators/?officeId=STL' } },
      },
      {
        id: 'STV',
        created: '2016-12-01T15:05:16.0000000Z',
        modified: '2020-02-14T15:52:02.0000000Z',
        name: 'Stevenage',
        manager: 'Evan Hunter',
        address: {
          buildingName: '',
          buildingNumber: '',
          line1: "St George's Way",
          line2: 'Stevenage',
          line3: 'Hertfordshire',
          line4: '',
          postcode: 'SG1 1XB',
          countryId: 'GB',
        },
        workPhone: undefined,
        email: 'stevenage.sales@hughmason.co.uk',
        metadata: {},
        _eTag: '"7B2C4CFDFFA0EE83468BABC0F4A8D63C"',
        _links: { self: { href: '/offices/STV' }, negotiators: { href: '/negotiators/?officeId=STV' } },
      },
    ],
    pageNumber: 1,
    pageSize: 3,
    pageCount: 3,
    totalCount: 25,
    _links: {
      self: { href: '/offices/?PageNumber=1&PageSize=3' },
      first: { href: '/offices/?PageNumber=1&PageSize=3' },
      next: { href: '/offices/?PageNumber=2&PageSize=3' },
      last: { href: '/offices/?PageNumber=9&PageSize=3' },
    },
  },
}

export const office: OfficeModel = {
  id: 'REA',
  created: '2020-02-28T10:34:06.0000000Z',
  modified: '2020-02-28T10:34:06.0000000Z',
  name: 'Reapit',
  manager: 'Mr John Smith',
  address: {
    buildingName: '',
    buildingNumber: '15',
    line1: 'Example street',
    line2: 'Solihull',
    line3: 'West Midlands',
    line4: '',
    postcode: 'B91 2XX',
    countryId: 'GB',
  },
  workPhone: '01234 567890',
  email: 'example@email.com',
  metadata: {},
  _eTag: '"104F6D31FAFEB3B1DE6BB9CF8E071094"',
  _links: { self: { href: '/offices/REA' }, negotiators: { href: '/negotiators/?officeId=REA' } },
}

export const mockCreateOfficeParams: CreateOfficeParams = {
  name: 'Reapit',
  address: {
    line1: 'London road',
    postcode: 'HP19 9QL',
  },
  workPhone: '0987654321',
  email: 'tester@reapit.com',
}

export const mockUpdateOfficeParams: UpdateOfficeParams = {
  id: 'REA',
  _eTag: '"104F6D31FAFEB3B1DE6BB9CF8E071094"',
  name: 'New Reapit',
}

export const mockChangeCellsForUpdateCase: ChangedCells = [
  {
    oldCell: {
      key: 'name',
      value: 'Reapit',
      isValidated: true,
    },
    newCell: {
      key: 'name',
      value: 'Reapit new name',
      isValidated: true,
    },
    row: 1,
    col: 0,
  },
]

export const mockChangeCellsForCreateCase: ChangedCells = [
  {
    oldCell: {
      key: 'name',
      value: '',
      isValidated: true,
    },
    newCell: {
      key: 'name',
      value: 'Reapit',
      isValidated: true,
    },
    row: 1,
    col: 0,
  },
]
