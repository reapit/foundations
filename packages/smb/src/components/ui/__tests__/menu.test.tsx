import React from 'react'
import { shallow } from 'enzyme'
import { getMockRouterProps } from '@/core/__mocks__/mock-router'
import { Menu, MenuProps, generateMenuConfig, callbackAppClick } from '@/components/ui/menu'
import { AuthContext } from '@/core/index'
import { mockContext } from '@/core/__mocks__/mock-context'

describe('Menu', () => {
  it('should match a snapshot', () => {
    const props = {
      ...getMockRouterProps({ params: {}, search: '' }),
    } as MenuProps
    const wrapper = shallow(
      <AuthContext.Provider value={mockContext}>
        <Menu {...props} />
      </AuthContext.Provider>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  describe('generateMenuConfig', () => {
    it('should return config', () => {
      const props = {
        ...getMockRouterProps({ params: {}, search: '' }),
      } as MenuProps
      const logoutCallback = jest.fn()
      const result = generateMenuConfig(logoutCallback, props.location, 'WEB')
      expect(result).toBeDefined()
    })
  })

  describe('callbackAppClick', () => {
    it('should run correcly', () => {
      const fn = callbackAppClick()
      expect(fn).toEqual('https://marketplace.reapit.cloud/client/installed')
    })
  })
})
