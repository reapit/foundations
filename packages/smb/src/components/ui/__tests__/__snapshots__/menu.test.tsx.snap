// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match a snapshot 1`] = `
<Memo(MenuComponent)
  defaultActiveKey="LOGO"
  location={
    Object {
      "hash": "mockHash",
      "key": "mockKey",
      "pathname": "mockPathname",
      "search": "",
      "state": Object {},
    }
  }
  menu={
    Array [
      Object {
        "icon": <Unknown
          className="nav-item-icon"
        />,
        "key": "LOGO",
        "type": "LOGO",
      },
      Object {
        "icon": <FaHome
          className="nav-item-icon"
        />,
        "key": "HOME",
        "title": "Home",
        "type": "PRIMARY",
        "url": "/home",
      },
      Object {
        "icon": <FaBuilding
          className="nav-item-icon"
        />,
        "key": "OFFICES",
        "title": "Offices",
        "type": "PRIMARY",
        "url": "/offices",
      },
      Object {
        "icon": <IoIosPeople
          className="nav-item-icon"
        />,
        "key": "USERS",
        "title": "Users",
        "type": "PRIMARY",
        "url": "/users",
      },
      Object {
        "icon": <FaClipboardList
          className="nav-item-icon"
        />,
        "key": "SOURCES",
        "title": "Sources",
        "type": "PRIMARY",
        "url": "/sources",
      },
      Object {
        "icon": <MdWeb
          className="nav-item-icon"
        />,
        "key": "MARKETING",
        "title": "Marketing",
        "type": "PRIMARY",
        "url": "/marketing",
      },
      Object {
        "icon": <MdLibraryBooks
          className="nav-item-icon"
        />,
        "key": "LETTINGS",
        "title": "Lettings",
        "type": "PRIMARY",
        "url": "/lettings",
      },
      Object {
        "icon": <MdLiveHelp
          className="nav-item-icon"
        />,
        "key": "SUPPORT",
        "title": "Support",
        "type": "PRIMARY",
        "url": "/support",
      },
      Object {
        "icon": <FaDownload
          className="nav-item-icon"
        />,
        "key": "INSTALL",
        "title": "Install",
        "type": "PRIMARY",
        "url": "/install",
      },
      Object {
        "callback": [Function],
        "icon": <FaCloud
          className="nav-item-icon"
        />,
        "key": "APPS",
        "title": "Apps",
        "type": "PRIMARY",
      },
      Object {
        "callback": [Function],
        "icon": <FaSignOutAlt
          className="nav-item-icon"
        />,
        "key": "LOGOUT",
        "title": "Logout",
        "type": "SECONDARY",
      },
    ]
  }
  mode="WEB"
/>
`;
